doc/jvm/xgboost4j_spark_tutorial.rst

```diff
@@ -0,0 +1,473 @@
+  |[5.0,3.6,1.4,0.2]|       0.0|[3.45569849014282...|[0.99579632282257...|       0.0|
+  |[5.4,3.9,1.7,0.4]|       0.0|[3.45569849014282...|[0.99428516626358...|       0.0|
+  |[4.6,3.4,1.4,0.3]|       0.0|[3.45569849014282...|[0.99643349647521...|       0.0|
+  |[5.0,3.4,1.5,0.2]|       0.0|[3.45569849014282...|[0.99579632282257...|       0.0|
+  |[4.4,2.9,1.4,0.2]|       0.0|[3.45569849014282...|[0.99618089199066...|       0.0|
+  |[4.9,3.1,1.5,0.1]|       0.0|[3.45569849014282...|[0.99636095762252...|       0.0|
+  |[5.4,3.7,1.5,0.2]|       0.0|[3.45569849014282...|[0.99428516626358...|       0.0|
+  |[4.8,3.4,1.6,0.2]|       0.0|[3.45569849014282...|[0.99643349647521...|       0.0|
+  |[4.8,3.0,1.4,0.1]|       0.0|[3.45569849014282...|[0.99618089199066...|       0.0|
+  |[4.3,3.0,1.1,0.1]|       0.0|[3.45569849014282...|[0.99618089199066...|       0.0|
+  |[5.8,4.0,1.2,0.2]|       0.0|[3.45569849014282...|[0.97809928655624...|       0.0|
+  |[5.7,4.4,1.5,0.4]|       0.0|[3.45569849014282...|[0.97809928655624...|       0.0|
+  |[5.4,3.9,1.3,0.4]|       0.0|[3.45569849014282...|[0.99428516626358...|       0.0|
+  |[5.1,3.5,1.4,0.3]|       0.0|[3.45569849014282...|[0.99579632282257...|       0.0|
+  |[5.7,3.8,1.7,0.3]|       0.0|[3.45569849014282...|[0.97809928655624...|       0.0|
+  |[5.1,3.8,1.5,0.3]|       0.0|[3.45569849014282...|[0.99579632282257...|       0.0|
+  +-----------------+----------+--------------------+--------------------+----------+
+
+Single instance prediction
+--------------------------
+
+XGBoostClassificationModel or XGBoostRegressionModel support make prediction on single instance as well.
+It accepts a single Vector as feature, and output the prediction label.
+
+However, the overhead of single-instance prediction is high due to the internal overhead of XGBoost, use it carefully!
+
+.. code-block:: scala
+
+  val features = xgbInput.head().getAs[Vector]("features")
+  val result = xgbClassificationModel.predict(features)
+
+Model Persistence 
+=================
+
+Model and pipeline persistence
+------------------------------
+
+A data scientist produces an ML model and hands it over to an engineering team for deployment in a production environment. Reversely, a trained model may be used by data scientists, for example as a baseline, across the process of data exploration. So it's important to support model persistence to make the models available across usage scenarios and programming languages.
+
+XGBoost4j-Spark supports saving and loading XGBoostClassifier/XGBoostClassificationModel and XGBoostRegressor/XGBoostRegressionModel. It also supports saving and loading a ML pipeline which includes these estimators and models.
+
+We can save the XGBoostClassificationModel to file system:
+
+.. code-block:: scala
+
+  val xgbClassificationModelPath = "/tmp/xgbClassificationModel"
+  xgbClassificationModel.write.overwrite().save(xgbClassificationModelPath)
```

Comment:
@hcho3 I think we may not need to explicitly say we can save in HDFS, as we rely on Spark's ML framework to do that (as here), so when user use Spark (with a proper hadoop configuration), it's automatically to be HDFS file

Reply:
I noticed that the user needs to compile with USE_HDFS=ON to use nativeBooster.saveModel(). Can we make a note? Or alternatively, we can alter the JARs in your maven repo to support HDFS out of box.

Reply:
good point, we may mention it in the section discussing how to interact across language bindings